==== .../wxMSW/2.8.3/src/common/event.cpp#1 - ...\wxMSW\2.8.3\src\common\event.cpp ====
@@ -1028,19 +1028,21 @@
 
     if (m_dynamicEvents)
     {
-        wxList::iterator it = m_dynamicEvents->begin(),
-                         en = m_dynamicEvents->end();
-        for (;it != en; ++it)
         {
+            wxList::iterator it = m_dynamicEvents->begin(),
+                            en = m_dynamicEvents->end();
+            for (;it != en; ++it)
+            {
 #if WXWIN_COMPATIBILITY_EVENT_TYPES
-            wxEventTableEntry *entry = (wxEventTableEntry*)*it;
+                wxEventTableEntry *entry = (wxEventTableEntry*)*it;
 #else // !WXWIN_COMPATIBILITY_EVENT_TYPES
-            wxDynamicEventTableEntry *entry = (wxDynamicEventTableEntry*)*it;
+                wxDynamicEventTableEntry *entry = (wxDynamicEventTableEntry*)*it;
 #endif // WXWIN_COMPATIBILITY_EVENT_TYPES/!WXWIN_COMPATIBILITY_EVENT_TYPES
 
-            if (entry->m_callbackUserData)
-                delete entry->m_callbackUserData;
-            delete entry;
+                if (entry->m_callbackUserData)
+                    delete entry->m_callbackUserData;
+                delete entry;
+            }
         }
         delete m_dynamicEvents;
     };

 	  	 
